pycape/attestation.py

```
@@ -28,7 +28,9 @@ def parse_attestation(attestation, root_cert):
     if public_key is None:
         raise ValueError("The public key is missing in the attestation document.")
 
-    return public_key
+    # Send back optional parameter user_data
+    user_data = doc.get("user_data")
```

we should raise an error if this key is missing

---

Actually, only run is returning this optional field, if we do cape test in the future then attestation won't have this field. Which is why the check happens in _run()

---

ahh, I see. I think this `parse_attestation` function is getting a bit overloaded, and anyway the name always seemed too broad. maybe we can break up the functionality a bit. but for now this is ok :+1: